### ISSUE TITLE
Chromium failed to generate and transfer data on RTCDataChannel

### ISSUE THREAD

**Issue by cpu82** [webrtc/samples#1102]
## Browser affected

Chromium 68.0.3440.106

## Description

It fails trying to generate and send data.

## Steps to reproduce

Go to https://webrtc.github.io/samples/src/content/datachannel/datatransfer/
Click on "Generate and send data"
(Ordered mode checkbox has been checked)

## Expected results

All the data has been sent.

## Actual results

The data has not been sent.

![screenshot at 2018-08-27 01-58-32](https://user-images.githubusercontent.com/4080230/44634722-acde2780-a99d-11e8-974b-8b91b9879e51.png)

**Comment by ErikHellman** [comment]
Thanks for report! I'm investigating an issue with bufferedAmountLow event not being fired. That would explain why this doesn't work anymore.

**Comment by cpu82** [comment]
Let me know if you need anything else to fix the issue.

Thanks

**Comment by ErikHellman** [comment]
@cpu82 If you are able to trigger a `bufferedAmountLow` event in your own project with a RTCDataChannel, please let me know. At the moment I'm not able to trigger it, which is why the sample fails.

**Comment by cpu82** [comment]
No, I'm using Chromium on FreeBSD. Is FreeBSD a supported OS? Can it be related to the problem?

**Comment by ErikHellman** [comment]
Probably not. I'm filed an issue no Chrome that you can track for this: https://bugs.chromium.org/p/chromium/issues/detail?id=878682

I'll update this sample to only use a polling solution until the event is emitted proprely.

**Comment by cpu82** [comment]
OK, I'll do some tests when you update the datatransfer sample.

**Comment by cpu82** [comment]
I get this result by running these [tests](http://w3c-test.org/webrtc/RTCDataChannel-bufferedAmount.html).

![screenshot at 2018-08-29 18-51-19](https://user-images.githubusercontent.com/4080230/44802887-23a53b80-abbd-11e8-8645-15d0a968c4a2.png)

**Comment by sumnulu** [comment]
```javascript
 while (sendProgress.value < sendProgress.max) {
      if (sendChannel.bufferedAmount > bufferFullThreshold) {
        if (usePolling) {
          setTimeout(sendAllData, 250);
        } else {
          sendChannel.addEventListener('bufferedamountlow', listener);
        }
      // ---- Missing break; ----
      } else {
        sendProgress.value += chunkSize;
        sendChannel.send(stringToSendRepeatedly);
      }
    }
```
https://github.com/webrtc/samples/blob/gh-pages/src/content/datachannel/datatransfer/js/main.js#L102

There is missing `break` statement at line 102, this is causing busy wait.

**Comment by ErikHellman** [comment]
@sumnulu Yes, if the event would be fired correctly that would be a problem. However, at the moment, the bufferedamountlow never fires in this sample. Check the PR I referenced above for details.

**Comment by sumnulu** [comment]
after the fix mine is working, maybe the "busy" part is preventing it to fire?
tested with "chrome Version 68.0.3440.106 (Official Build) (64-bit)"

**Comment by sumnulu** [comment]
@ErikHellman https://youtu.be/BoG3fbAeA3M

**Comment by ErikHellman** [comment]
I've merged #1112 which fixes this. This is not a perfect solution and we'll be monitoring the linked bugs in Chrome and will update this sample once this changes.
